app/views/exception_notifier/_session.text.erb

```diff
@@ -16,6 +16,8 @@ Referer: <%= @request.referer %>
 <% session['user_return_to'] = session['user_return_to']&.split('?')&.first %>
 Session: <%= session %>
 
-User UUID: <%= @kontroller.analytics_user.uuid %>
+<% user = @kontroller.analytics_user %>
+User UUID: <%= user.uuid %>
+User's Country (based on phone): <%= Phonelib.parse(user.phone).country %>
```

Comment:
@kontroller.analytics_user may be nil, so in some circumstances the exception notifier itself is exploding here

Reply:
If there is a controller, analytics_user cannot be `nil`. If there is no current_user, it is an instance of `AnonymousUser`. Perhaps what is happening is that @kontroller can be nil in some cases. Do you have a specific scenario?

Reply:
There won't be a controller if rails dies before serving a request. Encountered this in my environment.

Reply:
```
  ERROR: Failed to generate exception summary:

  ActionView::Template::Error: undefined method `uuid' for nil:NilClass

  /srv/idp/releases/chef/app/views/exception_notifier/_session.text.erb:19:in `_app_views_exception_notifier__session_text_erb__269420484499780557_47443798649360'
```

Reply:
Actually that message pretty strongly points to analytics_user being nil when the controller is present.

Reply:
In this case triggered by a redis connection failure.

```
A Redis::CannotConnectError occurred in #:

  Error connecting to Redis on redis.login.gov.internal:6379 (Errno::ECONNREFUSED)
  lib/utf8_sanitizer.rb:20:in `call'
```

Reply:
I see it now in the logs. Looks like it's happening before it gets to the controller. I will fix.

Reply:
👍 Thanks, no rush